tools/block-generator/generator/generate_test.go

```diff
@@ -229,17 +229,48 @@ func TestWriteRoundZero(t *testing.T) {
 func TestWriteRound(t *testing.T) {
 	partitiontest.PartitionTest(t)
 	g := makePrivateGenerator(t, 0, bookkeeping.Genesis{})
-	var data []byte
-	writer := bytes.NewBuffer(data)
-	g.WriteBlock(writer, 0)
-	g.WriteBlock(writer, 1)
-	var block rpcs.EncodedBlockCert
+
+	prepWriter := func() ([]byte, *bytes.Buffer, rpcs.EncodedBlockCert) {
+		var data []byte
+		writer := bytes.NewBuffer(data)
+		return data, writer, rpcs.EncodedBlockCert{}
+	}
+
+	// Initial conditions of g from makePrivateGenerator:
+	require.Equal(t, uint64(0), g.round)
+
+	data, writer, block := prepWriter()
+	err := g.WriteBlock(writer, 0)
+	require.NoError(t, err)
+
+	require.Equal(t, uint64(1), g.round)
+	protocol.Decode(data, &block)
+	require.Equal(t, block, rpcs.EncodedBlockCert{})
```

Comment:
why is block expected to be empty after data has be decoded to block?

Reply:
This was an observation of what was actually occuring in the test. I'm not aiming to assert that it should be empty in every possible situation. I'm also not wed to keeping this assertion so can happily remove if there is sufficient objection.

Reply:
I'm surprised that this assertion is true though. isn't block the genesis block? does genesis block actually equal to rpcs.EncodedBlockCert{}?

Reply:
I looked a bit closer at what's going on here. The genesis block -indeed- shouldn't be empty. Simply `data` is being used incorrectly. This is a preexisting bug in the test, but I hadn't looked closely enough to realize it.

Preparing a fix.

Reply:
see 027209ef2600c3488184f82e843d8768506deff3